text/0745-user-provided-cdk-init-templates.md

```diff
@@ -0,0 +1,626 @@
+   * Cons:
+     * Template authors must write JS-based generators (see a minimal example in Appendix B) instead of simple static file templates.
+     * Relies on the Yeoman runtime and ecosystem so CDK loses control over some UX and its ability to provide new features to template authors is limited
+
+While Yeoman offers strong scaffolding logic and built-in post-processing automation, adopting it as the default CDK template engine would couple our experience to an external runtime and ecosystem, reducing flexibility for future CLI features. The proposed solution retains those benefits for authors who choose Yeoman while allowing others to use any dynamic or manual approach, keeping CDK independent of a single tooling dependency.
+
+**Projen**
+
+Projen is a project configuration tool that defines all project settings, dependencies, and configuration files in a single TypeScript (or JavaScript) definition file (.projenrc.ts). Instead of manually editing generated files, users modify the Projen configuration, and Projen re-synthesizes the project structure.
+
+Project types in Projen are implemented as subclasses of Project (or a language-specific base project) and can be published as NPM packages so others can install them and initialize new projects. The cdk init command could integrate with Projen by invoking npx projen new <project-type>, optionally passing parameters to skip prompts or apply CDK defaults.
+
+Projen supports multi-language projects through specialized project types for Java, Python, C#, Go, and others, each with tailored file structures, dependency management, and post-processing automation. Authors can also define hybrid project types that generate multiple language components in a single repo.
+
+A minimal example of a custom project type is shown in Appendix C.
+
+   * Pros: 
+     * Controlled and maintained by AWS so it is easier to add requested features.
+     * Automates common setup tasks (such as linting, testing, dependency management, and publishing configurations) by generating them from a single Projen configuration file
+   * Cons: 
+     * Steep learning curve for template authors, who must understand Projen’s configuration model to create or modify templates.
+     * By default, Projen regenerates files on every `pj synth`, so direct edits are lost; changes must be made in the config. A `--eject` option exists to remove the Projen dependency after initialization, preserving the generated files while removing Projen’s automation benefits.
+
+Projen’s automation and multi-language support are compelling, but its `.projenrc`-driven model diverges from the direct-edit workflows many CDK users expect. The proposed solution keeps Projen optional:
+   * Templates can provide a post-initialization script (by the template author) that automatically runs `npx projen` after `cdk init`.
```

Comment:
How can templates provide a post-initialization script with the current approach?